### PR TITLE
Added dry-run mode in Compile command

### PR DESCRIPTION
**Pull request by cmaglie**
It's useful in IDE integrations if we want to run the build but don't want to copy the build artifact in the sketch folder.

### CODE REVIEW

File: cli/compile/compile.go [hunk 1]
```diff
@@ -47,6 +47,7 @@ var (
 	port               string   // Upload port, e.g.: COM10 or /dev/ttyACM0.
 	verify             bool     // Upload, verify uploaded binary after the upload.
 	exportFile         string   // The compiled binary is written to this file
+	dryRun             bool     // Use this flag to now write the output file
```
Comment (rsora) [inline cli/compile/compile.go]:
```suggestion
	dryRun             bool     // Use this flag to compile without generating the output file
```

File: cli/compile/compile.go [hunk 2]
```diff
@@ -67,6 +68,7 @@ func NewCommand() *cobra.Command {
 	command.Flags().BoolVar(&preprocess, "preprocess", false, "Print preprocessed code to stdout instead of compiling.")
 	command.Flags().StringVar(&buildCachePath, "build-cache-path", "", "Builds of 'core.a' are saved into this path to be cached and reused.")
 	command.Flags().StringVarP(&exportFile, "output", "o", "", "Filename of the compile output.")
+	command.Flags().BoolVarP(&dryRun, "dry-run", "n", false, "Perform the build but do not copy the compile output.")
```
Comment (rsora) [inline cli/compile/compile.go]:
```suggestion
	command.Flags().BoolVarP(&dryRun, "dry-run", "n", false, "Perform the build but do not copy the compile output file.")
```

**Comment by ianfixes** [comment]
> Perform the build but do not copy the compile output file

This means that `--dry-run` does in fact do the compilation?  What is the "compile output file"? Is it the compile logs or the compiled binary?

**Comment by ianfixes** [comment]
I think I can answer my own question -- it seems to prevent a directory called `build` from being copied into your Sketch directory.

**Comment by per1234** [comment]
@ianfixes note this: https://github.com/arduino/arduino-cli/issues/1051

**Pull request merged**